gcp_variant_transforms/transforms/merge_headers.py

```diff
@@ -120,14 +49,15 @@ def merge(self, first, second):
     if (not isinstance(first, vcf_header_io.VcfHeader) or
         not isinstance(first, vcf_header_io.VcfHeader)):
       raise NotImplementedError
-
     self._merge_header_fields(first.infos, second.infos)
     self._merge_header_fields(first.filters, second.filters)
     self._merge_header_fields(first.alts, second.alts)
     self._merge_header_fields(first.formats, second.formats)
     self._merge_header_fields(first.contigs, second.contigs)
 
   def _merge_header_fields(self, first, second):
+    #type: (Dict[str, OrderedDict[str, Any]],
+    #       Dict[str, OrderedDict[str, Any]]) -> None
```

Comment:
This is a great example of how adding types actually makes reading the code much easier; I just experienced the difference looking at the old and the new versions; thanks for gradually adding types to the code.

On that note: Is the type of the `OrderedDict` values `Any` or `Union[str, int]`? Because it seems to me that `resolve()` expects `Union[str, int]`.

Reply:
I think so, but not sure as it depends on what parser sends to us. But I can confirm for the attributes we care/resolve here (i.e type and num) it's either str or int. Do you want me to change it to Union[str, int]?

Reply:
If we were sure they are always `str` or `int` then yes I would have preferred that. If we are not sure, then I think we should fix the type for `FieldConflictResolver.resolve` function to be `Any` (instead of `Union[str, int]`) because you are sending these values to that function. Either choice is fine with me.

So this is an issue which actually should be picked by tools. For example in IntelliJ, it will actually give us warning if you submit the code in its current form. It is on me to add a tool for doing type checking as part of presubmit which I haven't found cycles to work on yet, sorry. We should not have this discussion in a review thread.

Reply:
Added Union[str, int] as we discussed offline. Confirmed PyVcf always return str or int. If this changes later on (using another parser), we don't care (doesn't hurt anyway).